### ISSUE TITLE
ffi fails to build via bundler on MacOS 10.8.5

### ISSUE THREAD

**Issue by mattmartini** [ffi/ffi#342]
**PROBLEM:**
bundler fails to build ffi and so my rails apps are broken in my development environment.
None of the posts that I've read have solutions that work for me.  I get the same error in a number of rails apps.

What other troubleshooting steps can I take?
How do I fix this issue?

**BACKGROUND:**
After an upgrade to Mountain Lion from Snow Leopard, Xcode upgrade, Clean Homebrew install, I did a clean install of rvm.  Now I am trying to recreate the gemsets used by my rails apps.  I did:
    gem install rails -v3.2.16 (to get the base rails gems installed)
    rvm gemset import ruby-1.9.3-p545@CLIENTNAME.gems (to import gems)
    bundle install (to install dependencies and resync gemfile.lock)

**ISSUE:**
bundle install fails with this error message:
An error occurred while installing ffi (1.9.3), and Bundler cannot continue.
Make sure that `gem install ffi -v '1.9.3'` succeeds before bundling.

The complete gem_make.out is here: https://gist.github.com/mattmartini/b2f8ad3b4e6c57325d38

The mkmf.log is here: https://gist.github.com/mattmartini/d3256cb52e5eafb2346b

The config.log is here: https://gist.github.com/mattmartini/69407e118e92cadda2b0

I would like to stay on ruby 1.9.3 as the Production servers are using this.

**OBSERVATIONS:**
One of the errors thrown is that 'C compiler cannot create executables'  - this is not true.  I can compile C programs and bundler built nokogiri without a problem.

I did just notice that libffi, which was installed (along with gcc46) by rvm, is keg only.  Would bundler be able to use the lib in creating the gem?

**ENVIRONMENT:**
Mac OS X 10.8.5 (Mountain Lion)
Xcode 5.0.2 (command line tools installed)
Homebrew 0.9.5 (recent fresh install)
rvm 1.25.25 (stable)
ruby 1.9.3-p545
libffi 3.0.13 (brew via rvm)
gcc46 (brew via rvm)

**REVIEWED:**
I reviewed a number of posts of similar issues, but none had a solution that worked for me, including:
https://github.com/ffi/ffi/issues/286
https://stackoverflow.com/questions/10575679/bundle-update-fails-on-ffi
https://stackoverflow.com/questions/7491606/cant-install-ffi-gem-rails-fail
https://stackoverflow.com/questions/11725577/an-error-occured-while-installing-ffi-1-1-2-and-bundler-cannot-continue

**Comment by cloudbring** [comment]
As a note, I was experiencing this error. The advice given was to use: 

`$ xcode-select -p` and look for the output 

`/Applications/Xcode.app/Contents/Developer`

When I tried to use `gcc` I got the following error:
```


Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.

```

After that, I had to:

`$ sudo gcc`

Which then asks me to sign a EULA for `clang` on the command-line before I can use `gcc` (which on OSX is an alias for `clang`).

Agreeing solved the problem when ti came to installing.

`$ gem install ffi -v '1.9.10'` worked fine
